docs/src/references/tedge-config-dir.md

```diff
@@ -0,0 +1,94 @@
+# Custom path for `thin-edge` config files
+
+The `tedge` CLI `--config-dir` option can be used to store the thin-edge
+configuration files in a custom directory.
+
+## Create all the directories
+Create all the directories that are needed by the thin-edge for its operations in a custom path as below.
```

Comment:
I do think that this doc should be a reference of the `--init` option and not of the `--config-dir` option. The later is here to overwrite the default location and what matters are the files and directories created. These are the specifications of the directory layout expected by thin-edge.

Reply:
`--init` is just one part of this document. The `--config-dir`  option is also used even for the other things like `config set/get`,  `cert create/remove`, and `connect`.

Reply:
This document is about the file layout created by `--init` and used by all `tedge`  sub-commands. The `--config-dir` option gives the ability to use a directory that is not the default `/etc/tedge` - nothing more.

I don't think you have a lot to change.

*  "`thin-edge` config files" would be a more appropriate title.
* The introduction should be about `--init` option to create the file layout expected by all thin-edge commands and daemons.
* The default directory path must be given and the `--config-dir` option cited as the way to override the default.